examples/esplora_backend.rs

```diff
@@ -0,0 +1,81 @@
+use std::str::FromStr;
+
+use bdk::{database::MemoryDatabase, blockchain::esplora::EsploraBlockchain, Wallet, KeychainKind, template::Bip84, SyncOptions, wallet::{AddressIndex, export::FullyNodedExport}};
+use bitcoin::{Network, util::bip32::{ExtendedPrivKey, self}};
+use bdk::blockchain::Blockchain;
+
+pub mod utils;
+
+use crate::utils::tx::build_signed_tx;
+
+/// This will create a wallet from an xpriv and get the balance by connecting to an Esplora server.
+/// If enough amount is available, this will send a transaction to an address.
+/// Otherwise, this will display a wallet address to receive funds.
+///
+/// This can be run with `cargo run --features="use-esplora-ureq" --example esplora_backend` in the root folder.
+///
+/// Users wanting asynchronous HTTP calls should change `required-features = ["use-esplora-ureq"]` to
+/// `required-features = ["use-esplora-reqwest"]` in `esplora_backend` target in Cargo.toml and run
+/// `cargo run --features="use-esplora-reqwest, reqwest-default-tls" --example esplora_backend`
+/// Without `reqwest-default-tls`, the asynchronous version will not work.
```

Comment:
This is not fully correct.. In order to use async calls they also need to use the `async-interface` feature along with `use-esplora-reqwest`.. And when they do that this same example will not work anymore.. For example you need to then call `wallet.sync().await.unwrap()` ..

So just changing the required feature doesn't work. if we need to show case the async reqwest thing we need to create another example..

What I would suggest here is use the async and reqwest in the example instead of ureq.. This example is very similar to the electrum one, so just doing another blocking esplora doesn't add much value.. Instead we should do a non blocking reqwest to show how to use the async-interface of bdk..

Reply:
If I run `wallet.sync(&blockchain, SyncOptions::default()).unwrap();` using `use-esplora-reqwest, reqwest-default-tls`, it runs without error, but the wallet shows empty balance because apparently it does not sync.

Reply:
I think you also need to pass the `async-interface` feature.. Thats what makes the bdk reqwest codes async.. The error seems to be indicating that.. Its getting a regular `Result<>` whereas it expected a `Future`..

Reply:
I had removed `async-interface` due to error below.
It is weird because `esplora_backend` doesn't have any reference to `electrum` .

```bash
$ cargo run --features="use-esplora-reqwest, reqwest-default-tls, async-interface" --example esplora_backend

   Compiling bdk v0.20.1-dev (/home/node01/Dev/wbdk)
error: Features async-interface and electrum are mutually exclusive and cannot be enabled together
   --> src/lib.rs:214:1
    |
214 | / compile_error!(
215 | |     "Features async-interface and electrum are mutually exclusive and cannot be enabled together"
216 | | );
    | |_^
```

Reply:
I think getting `reqwest` working is not always straight forward and might need some trial and errors.. I think these examples are still very useful without `reqwest` and with `ureq` instead.. That might help us move this one forward, and do the async example in a independent PR. 

How does that sound to you @w0xlt ??

Reply:
I think it is working now.
Apparently https://github.com/bitcoindevkit/bdk/pull/526/commits/400cbad4388322f4f0e0007365bf1a68ea95a6b0 is working with `reqwest`.